scene/main/canvas_item.cpp

```diff
@@ -489,6 +522,8 @@ void CanvasItem::set_as_top_level(bool p_top_level) {
 	if (get_viewport()) {
 		get_viewport()->canvas_item_top_level_changed();
 	}
+
+	update_configuration_warnings();
```

Comment:
The child items' warnings might be affected/changed as a result. Does this propagate the update for the children? :thinking:

Reply:
It does! Tested out, changing the mask or the top level status of a parent node updates the check of children

Reply:
Hmm, I think it doesn't actually say the children to update their warnings, it's just this single update. So AFAICT technically the answer to my question is no.

But the warnings in the editor's SceneTree dock are updated for the children correctly because it handles the `SceneTree.node_configuration_warning_changed(Node)` signal by updating the whole tree, and thus fetching all warnings anew. So it "just works" nonetheless.

Anyway, I think all / almost all warnings are updated like in here and it's good enough / not an issue because of the above.

Reply:
Can move it to the propagation check and see

Reply:
That would be more correct/reliable, yes. But I see that e.g. in CanvasModulate I did it as incorrectly as you in here :upside_down_face:, aka notified only about the one being changed / the one triggering the change, instead of notifying about all which are affected / have their warning changed.

Again, for the in-engine usage it seems it currently doesn't matter, as in the editor the whole scene tree / all Tree items are being recreated anyway. If we'd want to change it to only update relevant items in the Tree then this would became a real issue indeed.

From the user standpoint it might be surprising though. However it seems not reported, aka only a theoretical issue? :thinking:

I'd say let's keep it as is in here, and if we want to correct this then it would make sense to do so in a separate PR by checking/fixing all `update_configuration_warnings()` accross the codebase (no idea how many of them are in a similar manner not notifying about other affected nodes).

Reply:
I'll add a note that it relies on the tree being updated for now

Reply:
For the top level part there's a propagated method I'll add the check to instead, but I'll add a note to the layer one rather than writing a new propagation for it